homeassistant/components/tplink/light.py

```diff
@@ -1,552 +1,144 @@
 """Support for TPLink lights."""
 from __future__ import annotations
 
-import asyncio
-from collections.abc import Mapping
-from datetime import timedelta
 import logging
-import re
-import time
-from typing import Any, NamedTuple, cast
-
-from pyHS100 import SmartBulb, SmartDeviceException
+from typing import Any
 
 from homeassistant.components.light import (
     ATTR_BRIGHTNESS,
     ATTR_COLOR_TEMP,
     ATTR_HS_COLOR,
-    SUPPORT_BRIGHTNESS,
-    SUPPORT_COLOR,
-    SUPPORT_COLOR_TEMP,
+    ATTR_TRANSITION,
+    COLOR_MODE_BRIGHTNESS,
+    COLOR_MODE_COLOR_TEMP,
+    COLOR_MODE_HS,
+    COLOR_MODE_ONOFF,
+    SUPPORT_TRANSITION,
     LightEntity,
 )
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.core import HomeAssistant
-from homeassistant.exceptions import HomeAssistantError, PlatformNotReady
-import homeassistant.helpers.device_registry as dr
-from homeassistant.helpers.entity import DeviceInfo
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 from homeassistant.util.color import (
     color_temperature_kelvin_to_mired as kelvin_to_mired,
     color_temperature_mired_to_kelvin as mired_to_kelvin,
 )
-import homeassistant.util.dt as dt_util
 
-from . import CONF_LIGHT, DOMAIN as TPLINK_DOMAIN
-from .common import add_available_devices
+from .const import DOMAIN
+from .coordinator import TPLinkDataUpdateCoordinator
+from .entity import CoordinatedTPLinkEntity, async_refresh_after
 
 PARALLEL_UPDATES = 0
```

Comment:
```suggestion
```

We can drop this now since its the default. https://developers.home-assistant.io/docs/integration_fetching_data/#request-parallelism